File: 백준/Bronze/31712. 핑크빈 레이드/핑크빈 레이드.c
#include <stdio.h>

int main() {
    int Cu, Du, Cd, Dd, Cp, Dp, H;
    int sec = 0;

    scanf("%d %d", &Cu, &Du);
    scanf("%d %d", &Cd, &Dd);
    scanf("%d %d", &Cp, &Dp);
    scanf("%d", &H);

    while (H > 0) {
        if (sec % Cu == 0) {
            H -= Du;
        }
        if (sec % Cd == 0) {
            H -= Dd;
        }
        if (sec % Cp == 0) {
            H -= Dp;
        }
        if (H <= 0) {
            break;
        }
        sec++;
    }
    printf("%d\n", sec);

    return 0;
}

File: 백준/Bronze/31712. 핑크빈 레이드/핑크빈 레이드.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    Cu, Du = map(int, input().split())
    Cd, Dd = map(int, input().split())
    Cp, Dp = map(int, input().split())
    H = int(input())
    sec = 0
    while 1:
        if sec%Cu==0:
            H -= Du
        if sec%Cd==0:
            H -= Dd
        if sec%Cp==0:
            H -= Dp
        if H <= 0:
            break
        sec += 1
    print(sec)


if __name__ == '__main__':
    main()